packages/sdk/src/evm/core/classes/erc-20-history.ts

```diff
@@ -55,14 +55,24 @@ export class TokenERC20History {
         balances[to] = balances[to].add(amount);
       }
     });
+    const _promises = Object.entries(balances).map(async ([addr, value]) => {
+      const result = await fetchCurrencyValue(
+        this.contractWrapper.getProvider(),
+        this.contractWrapper.address,
+        value,
+      );
+      return { [addr]: result };
+    });
+    const balanceData = (await Promise.all(_promises)).reduce(
+      (result, currentObject) => {
+        return { ...result, ...currentObject };
+      },
+      {},
+    );
     return Promise.all(
       Object.keys(balances).map(async (addr) => ({
         holder: addr,
-        balance: await fetchCurrencyValue(
-          this.contractWrapper.getProvider(),
-          this.contractWrapper.address,
-          balances[addr],
-        ),
+        balance: balanceData[addr],
```

Comment:
this `map` doesnt need to be async anymore right?

Reply:
Oops you're right

Reply:
This code looks off. I'm working on it

Reply:
@joaquim-verges would you be able to take a look at the code again? Thank you